packages/vscode-extension/webpack.config.js

```diff
@@ -94,4 +95,136 @@ const config = {
     level: 'log', // enables logging required for problem matchers
   },
 };
-module.exports = [config];
+
+const browserConfig = {
+  target: 'webworker',
+  entry: {
+    extension: './src/browser/extension.ts',
+  },
+  output: {
+    path: path.resolve(__dirname, 'dist', 'browser'),
+    filename: '[name].js',
+    libraryTarget: 'commonjs',
+    devtoolModuleFilenameTemplate: '../../[resource-path]',
+  },
+  context: path.resolve(__dirname),
+  externals: {
+    vscode: 'commonjs vscode', // the vscode-module is created on-the-fly and must be excluded
+    prettier: 'commonjs ./prettier',
+  },
+  resolve: {
+    // support reading TypeScript and JavaScript files, 📖 -> https://github.com/TypeStrong/ts-loader
+    extensions: ['.ts', '.js'],
+    plugins: [new TsconfigPathsPlugin({})],
+  },
+  optimization: {
+    minimize: false,
+  },
+  module: {
+    parser: {
+      javascript: {
+        commonjsMagicComments: true,
+      },
+    },
+    rules: [
+      {
+        test: /\.ts$/,
+        exclude: /node_modules/,
+        use: {
+          loader: 'ts-loader',
+          options: {
+            projectReferences: true,
+          },
+        },
+      },
+    ],
+  },
+  devtool: 'nosources-source-map',
+  plugins: [
+    new webpack.DefinePlugin({
+      // A flag that lets us do fun webpack-only shenanigans...
+      'process.env.WEBPACK_MODE': true,
+    }),
+  ],
+  node: {
+    __filename: 'eval-only', // this means that __filename will eval to the output file name
+    __dirname: 'eval-only', // this means that __dirname will eval to the dist folder
+  },
+  infrastructureLogging: {
+    level: 'log', // enables logging required for problem matchers
+  },
+};
+
+const browserServerConfig = async () => {
+  const docsManager = new ThemeLiquidDocsManager();
+  const [tags, filters, objects, systemTranslations, schemas] = await Promise.all([
+    docsManager.tags(),
+    docsManager.filters(),
+    docsManager.objects(),
+    docsManager.systemTranslations(),
+    docsManager.schemas('theme'),
+  ]);
+
+  return {
+    context: path.resolve(__dirname),
+    target: 'webworker',
+    mode: 'none',
+    entry: {
+      server: './src/browser/server.ts',
+    },
+    output: {
+      path: path.resolve(__dirname, 'dist', 'browser'),
+      filename: '[name].js',
+      libraryTarget: 'var',
+      library: 'serverExportVar',
+      devtoolModuleFilenameTemplate: '../../[resource-path]',
```

Comment:
The only reason I need 2x a config that should look like one is because of this difference here. The worker has to be "import-less", but the extension needs a `require('vscode')` call in the output that gets shimmed.